packages/@lwc/template-compiler/src/shared/types.ts

```diff
@@ -4,132 +4,204 @@
  * SPDX-License-Identifier: MIT
  * For full license text, see the LICENSE file in the repo root or https://opensource.org/licenses/MIT
  */
-import * as parse5 from 'parse5';
 import { CompilerDiagnostic } from '@lwc/errors';
 
-export type TemplateIdentifier = { type: 'Identifier'; name: string };
-export type TemplateExpression =
-    | {
-          type: 'MemberExpression';
-          object: TemplateExpression;
-          property: TemplateExpression;
-          computed: boolean;
-          optional: boolean;
-      }
-    | { type: 'Literal'; value: string | number | boolean | null }
-    | TemplateIdentifier;
-
 export type TemplateCompileResult = {
     code: string;
     warnings: CompilerDiagnostic[];
 };
 
 export type TemplateParseResult = {
-    root?: IRElement | undefined;
+    root?: Root | undefined;
     warnings: CompilerDiagnostic[];
 };
 
-export interface ForEach {
-    expression: TemplateExpression;
-    item: TemplateIdentifier;
-    index?: TemplateIdentifier;
+export enum LWCDirectiveDomMode {
+    manual = 'manual',
 }
 
-export interface ForIterator {
-    expression: TemplateExpression;
-    iterator: TemplateIdentifier;
+export enum LWCNodeType {
```

Comment:
I wasn't sure if it was a good idea to use an enum here, my thought process was it would be easier to keep the string definitions in one place but I see everything else on the template compiler is using strings.  I can change these to string values if we prefer strings instead.

Reply:
I would keep the string inline for now.